common/client-core/src/client/mix_traffic/transceiver.rs

```diff
@@ -123,6 +138,20 @@ where
     fn ws_fd(&self) -> Option<RawFd> {
         self.gateway_client.ws_fd()
     }
+
+    async fn send_client_request(
+        &mut self,
+        message: ClientRequest,
+    ) -> Result<(), GatewayClientError> {
+        if let Some(shared_key) = self.gateway_client.shared_key() {
+            self.gateway_client
+                .send_websocket_message(message.encrypt(shared_key.unwrap_legacy())?)
```

Comment:
why are you attempting to unwrap the legacy shared key here? any new-ish client will be failing here. just use `message.encrypt(shared_key)` (because it will then correctly delegate to correct enum variant)

Reply:
Got tripped up by Arc here